packages/gestalt/src/Upsell.js

```diff
@@ -154,15 +157,21 @@ export default function Upsell({
             </Box>
           </Box>
         </Box>
-        <Box smDisplay="flex" marginStart="auto" smMarginEnd={4} smPaddingY={3}>
-          {secondaryAction && responsiveMinWidth !== 'xs' && (
-            <UpsellAction type="secondary" data={secondaryAction} />
-          )}
-          {primaryAction && <UpsellAction type="primary" data={primaryAction} />}
-          {secondaryAction && responsiveMinWidth === 'xs' && (
-            <UpsellAction type="secondary" data={secondaryAction} stacked={!!secondaryAction} />
-          )}
-        </Box>
+        {children ? (
+          <Box smDisplay="flex" marginStart="auto" smMarginEnd={4} smPaddingY={3}>
```

Comment:
This Box is identical, so the ternary should be moved to inside of it to prevent the repetition (use a Fragment in the !children path)